File: apps/mobile/modules/pathwayos-terminal/ios/PathwayOSTerminalModule.swift
import ExpoModulesCore

public class PathwayOSTerminalModule: Module {
  public func definition() -> ModuleDefinition {
    Name("PathwayOSTerminalSurface")

    View(PathwayOSTerminalView.self) {
      Prop("terminalKey") { (view: PathwayOSTerminalView, terminalKey: String) in
        view.terminalKey = terminalKey
      }

      Prop("initialBuffer") { (view: PathwayOSTerminalView, initialBuffer: String) in
        view.initialBuffer = initialBuffer
      }

      Prop("fontSize") { (view: PathwayOSTerminalView, fontSize: Double) in
        view.fontSize = CGFloat(fontSize)
      }

      Prop("appearanceScheme") { (view: PathwayOSTerminalView, appearanceScheme: String) in
        view.appearanceScheme = appearanceScheme
      }

      Prop("themeConfig") { (view: PathwayOSTerminalView, themeConfig: String) in
        view.themeConfig = themeConfig
      }

      Prop("backgroundColor") { (view: PathwayOSTerminalView, backgroundColor: String) in
        view.backgroundColorHex = backgroundColor
      }

      Prop("foregroundColor") { (view: PathwayOSTerminalView, foregroundColor: String) in
        view.foregroundColorHex = foregroundColor
      }

      Prop("mutedForegroundColor") { (view: PathwayOSTerminalView, mutedForegroundColor: String) in
        view.mutedForegroundColorHex = mutedForegroundColor
      }

      Events("onInput", "onResize")
    }
  }
}

File: apps/mobile/modules/pathwayos-terminal/android/src/main/java/expo/modules/pathwayosterminal/PathwayOSTerminalModule.kt
package expo.modules.pathwayosterminal

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class PathwayOSTerminalModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("PathwayOSTerminalSurface")

    View(PathwayOSTerminalView::class) {
      Prop("terminalKey") { view: PathwayOSTerminalView, terminalKey: String ->
        view.terminalKey = terminalKey
      }

      Prop("initialBuffer") { view: PathwayOSTerminalView, initialBuffer: String ->
        view.initialBuffer = initialBuffer
      }

      Prop("fontSize") { view: PathwayOSTerminalView, fontSize: Double ->
        view.fontSize = fontSize.toFloat()
      }

      Prop("appearanceScheme") { view: PathwayOSTerminalView, appearanceScheme: String ->
        view.appearanceScheme = appearanceScheme
      }

      Prop("themeConfig") { view: PathwayOSTerminalView, themeConfig: String ->
        view.themeConfig = themeConfig
      }

      Prop("backgroundColor") { view: PathwayOSTerminalView, backgroundColor: String ->
        view.backgroundColorHex = backgroundColor
      }

      Prop("foregroundColor") { view: PathwayOSTerminalView, foregroundColor: String ->
        view.foregroundColorHex = foregroundColor
      }

      Prop("mutedForegroundColor") { view: PathwayOSTerminalView, mutedForegroundColor: String ->
        view.mutedForegroundColorHex = mutedForegroundColor
      }

      Events("onInput", "onResize")
    }
  }
}
